zstd-safe/src/lib.rs

```diff
@@ -81,6 +81,20 @@ pub const FRAMEHEADERSIZE_MIN: u32 = zstd_sys::ZSTD_FRAMEHEADERSIZE_MIN;
 pub const FRAMEHEADERSIZE_MAX: u32 = zstd_sys::ZSTD_FRAMEHEADERSIZE_MAX;
 pub const JOBSIZE_MIN: u32 = zstd_sys::ZSTDMT_JOBSIZE_MIN;
 
+type SafeResult = Result<usize, u32>;
+
+/// Parse the result code
+///
+/// Returns the number of bytes written if the code represents success,
+/// or the error message code otherwise.
+fn parse_code(code: usize) -> SafeResult {
+    let error_code = unsafe { zstd_sys::ZSTD_isError(code) as u32 };
```

Comment:
I think `ZSTD_isError` just returns a "boolean" (a 1 or 0 integer), and not the code itself. C doesn't have booleans, which is why `ZSTD_isError` returns a "unsigned integer".
You probably should use `zstd_safe::is_error` which returns a `bool` and is less confusing.

```rust
type SafeResult = Result<usize, usize>; // Why convert to `u32`?

fn parse_code(code: usize) -> SafeResult {
    if is_error(code) == 0 {
        Ok(code)
    } else {
        Err(code)
    }
} 
```

Reply:
True, I pushed more fixed and more clear solution.